runtime/src/tiered_storage/footer.rs

```diff
@@ -0,0 +1,331 @@
+use {
+    crate::tiered_storage::{file::TieredStorageFile, mmap_utils::get_type},
+    memmap2::Mmap,
+    serde::{Deserialize, Serialize},
+    solana_sdk::hash::Hash,
+    std::{mem, path::Path},
+};
+
+pub const FOOTER_FORMAT_VERSION: u64 = 1;
+
+// The size of the footer struct + the u64 magic number at the end.
+pub const FOOTER_SIZE: i64 = (mem::size_of::<TieredStorageFooter>() + mem::size_of::<u64>()) as i64;
```

Comment:
A static check sounds like a good idea as this will be the default footer size instead (meaning the size of the footer of the current build).

As we add new blocks like filter blocks (e.g. for bloom filters), the footer size of the footer block will change.  The change shouldn't be done often, and this also means there may be files that have different footer sizes (e.g., snapshots from other nodes or version upgrades but keep the local snapshot).